src/post/post.service.ts

```diff
@@ -499,4 +501,21 @@ export class PostService {
     async getOutboxItemCount(accountId: number): Promise<number> {
         return this.postRepository.getOutboxItemCount(accountId);
     }
+
+    async deleteByApId(
+        apId: URL,
+        account: Account,
+    ): Promise<Result<boolean, DeletePostError>> {
+        const postResult = await this.getByApId(apId);
+        if (isError(postResult)) {
+            return postResult;
+        }
+        const post = getValue(postResult);
+        if (post.author.id !== account.id) {
```

Comment:
Works either way, but just noting that we've changed this author check from a `uuid` check to `id` check